qiita_pet/static/js/qiita.js

```diff
@@ -376,55 +376,120 @@ function show_alert(data) {
   var proc_data_table = '<table class="table" cellpadding="0" cellspacing="0" border="0" style="padding-left:0px;width:95%">';
+  var processing_method = {};
   proc_data_table += '<tr>';
   if (for_study_list) {
     proc_data_table += '<th></th>';
+    proc_data_table += '<th>Artifacts</th>';
   }
-  proc_data_table += '<th>Name</th>';
+  proc_data_table += '<th>Processing method</th>';
   if (for_study_list) {
     proc_data_table += '<th>Data type</th>';
   }
-  proc_data_table += '<th>Processing method</th>';
   proc_data_table += '<th>Parameters</th>';
-  if (for_study_list) {
-    proc_data_table += '<th>Samples in Prep Info</th>';
-  }
-  proc_data_table += '<th>Files</th>';
   proc_data_table += '</tr>';
 
+  // grouping by processing_method, data_type and parameters
   $.each(data, function (idx, info) {
     if (typeof info !== 'string' && !(info instanceof String)) {
-      proc_data_table += '<tr>';
-      if (for_study_list) {
-        if (samples === null) {
-          proc_data_table += '<td><input type="button" class="btn btn-sm" value="Add" onclick="send_samples_to_analysis(this, [' + info.artifact_id + '])"></td>';
-        } else {
-          proc_data_table += '<td><input type="button" class="btn btn-sm" value="Add" onclick="send_samples_to_analysis(this, [' + info.artifact_id + "], '" + samples[info.artifact_id].join(',') + "'" + ')"></td>';
-        }
+      var algorithm = info.algorithm;
+      if (!(algorithm in processing_method)) {
+        processing_method[algorithm] = {};
       }
-      proc_data_table += '<td>' + info.name + ' (' + info.artifact_id + ' - ' + info.timestamp.split('.')[0] + ')</td>';
 
-      if (for_study_list) {
-        proc_data_table += '<td>' + info.data_type + ' (' + info.target_subfragment.join(', ') + ')</td>';
+      var data_type = info.data_type + ' (' + info.target_subfragment.join(', ') + ')';
+      if (!(data_type in processing_method[algorithm])) {
+        processing_method[algorithm][data_type] = {};
       }
-      proc_data_table += '<td>' + info.algorithm + '</td>';
 
       var params = '';
       for (var key in info.parameters) {
         params += '<i>' + key + '</i>: ' + info.parameters[key] + '<br/>';
       }
-      proc_data_table += '<td><small>' + params + '</small></td>';
-      if (for_study_list) {
-        proc_data_table += '<td>' + info.prep_samples + '</td>';
+      if (!(params in processing_method[algorithm][data_type])) {
+        processing_method[algorithm][data_type][params] = [];
       }
-      proc_data_table += '<td><small>' + info.files.join('<br/>')  + '</small></td>';
 
-      proc_data_table += '</tr>';
+      processing_method[algorithm][data_type][params].push(info);
     }
   });
 
+  // creating rows
+  $.each(processing_method, function (pm, data_types) {
+    $.each(data_types, function (dt, parameters) {
+      $.each(parameters, function (params, artifacts) {
+        proc_data_table += '<tr>';
```

Comment:
So this is going to add a new row that condenses all preps from a study that have the same unique combination of data type and parameters, right? I.e. all the artifacts added when clicking the 'all' button are going to be deblur OR CR, but not a mix of both?

Reply:
That's correct but just to be clear, it's gonna group by processing method (cr, deblur, etc), data type (16S, 18S, etc) and parameters (cr against gg, silva, etc)